### PR TITLE
add test infrastructure

### PR DESCRIPTION
**Pull request by dduugg**

### CODE REVIEW

File: server/config/helpers.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
  * Handle errors by responding with 500 and the error message.
  * @param {Object} error the error object
  * @param {Object} request the http ClientRequest object
- * @param {Object} response the htpp ServerResponse object
+ * @param {Object} response the http ServerResponse object
```
Comment (zindlerb) [inline server/config/helpers.js]: :open_mouth:

**Comment by zindlerb** [comment]
:+1:

File: test/server/config/helpersSpec.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    });
+  });
+
+  describe("getAWSCallbackHandler", function() {
+    it("returns a callback that handles data", function(done) {
+      supertest(mockApp)
+        .get("/awsData")
+        .expect(200)
+        .expect(MOCK_DATA)
+        .end(done);
```
Comment (owen-d) [inline test/server/config/helpersSpec.js]: Does this work as expected? According to the documentation:

If you are using the .end() method .expect() assertions that fail will not throw - they will return the assertion as an error to the .end() callback. In order to fail the test case, you will need to rethrow or pass err to done()

Comment (dduugg) [inline test/server/config/helpersSpec.js]: It works as I expect. For instance, when I change `200` to `201`, the test fails:

    Error: expected 201 "Created", got 200 "OK"

What is your expectation?

**Pull request merged**